### PR TITLE
Performance Refactor for `privacy-experience`

### LINKED ISSUE #4040: Add "real" endpoints to the drill bencharks
### Is your feature request related to a specific problem?

We want to test endpoints on both the ops and ctl side, instead of only the simple "health" ones.

### Describe the solution you'd like

For authentication, it looks like we can hit the `login` endpoint first and use the response body in a downstream request

### LINKED ISSUE #4171: Refactor the GET `privacy-experience` endpoint to be more performant
### Is your feature request related to a specific problem?

_A description of what the problem is._
_Ex. I'm always frustrated when [...]_

### Describe the solution you'd like

_A description of what you want to happen._

### Describe alternatives you've considered, if any

_A description of any alternative solutions or features you've considered._

### Additional context

_Add any other context or screenshots about the feature request here._

### PR DESCRIPTION
**Pull request by ThomasLaPiana**
Closes #4040 

### Description Of Changes

See below ⬇️ 

This PR will be followed up with another that focuses on adding proper Caching to that endpoint

### Code Changes

* [x] update `drill` to give us more data for A/B testing
* [x] sprinkle in `asyncio.sleep` calls to force `privacy-experience` to give up the event loop between heavy db calls

### Steps to Confirm

* [ ] run `nox -s dev` and then `nox -s load_tests`, it should be observed that the response times for all endpoints (besides `privacy-experience`) are extremely consistent, representing that they aren't being significantly blocked by `privacy-experience`

### Pre-Merge Checklist

* [ ] All CI Pipelines Succeeded
* Documentation:
  * [ ] documentation complete, [PR opened in fidesdocs](https://github.com/ethyca/fidesdocs/pulls)
  * [ ] documentation [issue created in fidesdocs](https://github.com/ethyca/fidesdocs/issues/new/choose)
* [ ] Issue Requirements are Met
* [ ] Relevant Follow-Up Issues Created
* [ ] Update `CHANGELOG.md`
* [ ] For API changes, the [Postman collection](https://github.com/ethyca/fides/blob/main/docs/fides/docs/development/postman/Fides.postman_collection.json) has been updated

### CODE REVIEW

**Comment by ThomasLaPiana** [comment]
I found a good caching library but it requires a newer version of the Python Redis library, which is a breaking change and will require updates on our side...going to give it a shot

File: requirements.txt [hunk 1]
```diff
@@ -11,7 +11,6 @@ deepdiff==6.3.0
 defusedxml==0.7.1
 expandvars==0.9.0
 fastapi[all]==0.89.1
-fastapi-caching[redis]==0.3.0
```
Comment (ThomasLaPiana) [inline requirements.txt:14]: this was apparently unused, and the package itself is completely abandoned

File: src/fides/api/api/v1/endpoints/privacy_experience_endpoints.py [hunk 2]
```diff
@@ -174,6 +175,7 @@ def privacy_experience_list(
             db=db, fides_user_device_id=fides_user_device_id
         )
 
+    await asyncio.sleep(delay=0.001)
```
Comment (pattisdr) [inline src/fides/api/api/v1/endpoints/privacy_experience_endpoints.py:178]: Interesting, where did you come up with this trick?

**Review by pattisdr — APPROVED** [review]

File: src/fides/api/api/v1/endpoints/privacy_experience_endpoints.py (hunk 2 above)
Comment (ThomasLaPiana) [reply src/fides/api/api/v1/endpoints/privacy_experience_endpoints.py:178]: I recall having to use this trick years ago but don't really remember where I learned it...probably from a book like `Fluent Python` if I had to guess

**Pull request merged**